### DISCUSSION TITLE
usage outside of setup

### DISCUSSION THREAD

**Discussion by xvaara** [Akryum/vue-apollo#1410] [Q&A] [answered]
so I have a codebase with graphql-codegen which wraps the queries in functions...
 ```
const { resolveClient } = useApolloClient();
      const client = resolveClient();

provideApolloClients({ myCustomClientId: client })(() => {
    const { onResult: onCurrentuserResult } = useCurrentUserUpdatedSubscription(() => ({
      enabled: this.$state.loggedIn,
      clientId: "myCustomClientId"
    }));
....
})
```
and i get `Error: Apollo client with id myCustomClientId not found. Use provideApolloClient() if you are outside of a component setup.`

It's inside a pinia action. It worked with 4.0.0-alpha.16 without the provideApolloClients in pinia, but with .19 it complains the same but with id default.

How is this supposed to work now?

I've downgraded to alpha.16 for now.

**Comment by dominikklein** [comment]
Normally it should work outside of a setup also with the current version.

We use the `provideApolloClient` during the app initialization and afterward, we have some pinia stores which are using queries.

**Comment by xvaara** [comment, marked as answer]
Classical pebkac-problem. I didn't check that I upgraded the module also in a different package in my yarn workspace. So the problem was that build included two versions, and they conflicted. It worked with dev server, so I was confused and miffed it didn't work in production.
